### PR TITLE
Allow skipping of sourceDirectory or testSourceDirectory

### PR DESCRIPTION
**Pull request by camac**
Fixes spotify/fmt-maven-plugin#31

Adds 2 extra configuration options controlling whether to skip relevant
directories:

- skipSourceDirectory
- skipTestSourceDirectory

Tests added:

- check_skipsourcedirectory: Tests that sourceDirectory is skipped
  for check goal when skipSourceDirectory is true.
  sourceDirectory includes unformatted code so it would fail if not
  skipped.
- check_skiptestsourcedirectory: Tests that testSourceDirectory is
  skipped for check goal when skipTestSourceDirectory is true.
  testSourceDirectory includes unformatted code so it would
  fail if not skipped.
- skipsourcedirectory: Tests that sourceDirectory files are not formatted
  with format goal when skipSourceDirectory is true.
- skiptestsourcedirectory: Tests that testSourceDirectory files are not
  formatted with format goal when skipTestSourceDirectory is true.

### CODE REVIEW

**Comment by klaraward** [comment]
Thanks @camac :)

Can you also add `invoker.properties` files for these tests? Might not be the most interesting thing to test via the invoker test mechanism but still nice to keep things consistent.

File: src/test/java/com/spotify/fmt/FMTTest.java [hunk 1]
```diff
@@ -30,6 +30,14 @@ public void noSource() throws Exception {
     assertThat(fmt.getResult().processedFiles()).isEmpty();
   }
 
+  @Test
+  public void skipSource() throws Exception {
+    FMT fmt = (FMT) mojoRule.lookupConfiguredMojo(loadPom("skipsourcedirectory"), FORMAT);
+    fmt.execute();
+
+    assertThat(fmt.getFilesProcessed()).hasSize(1);
```
Comment (klaraward) [inline src/test/java/com/spotify/fmt/FMTTest.java:38]:
```suggestion
    assertThat(fmt.getResult().processedFiles()).hasSize(1);
```

File: src/test/java/com/spotify/fmt/FMTTest.java [hunk 2]
```diff
@@ -38,6 +46,14 @@ public void withoutTestSources() throws Exception {
     assertThat(fmt.getResult().processedFiles()).hasSize(2);
   }
 
+  @Test
+  public void skipTestSources() throws Exception {
+    FMT fmt = (FMT) mojoRule.lookupConfiguredMojo(loadPom("skiptestsourcedirectory"), FORMAT);
+    fmt.execute();
+
+    assertThat(fmt.getFilesProcessed()).hasSize(2);
```
Comment (klaraward) [inline src/test/java/com/spotify/fmt/FMTTest.java:54]:
```suggestion
    assertThat(fmt.getResult().processedFiles()).hasSize(2);
```

**Comment by camac** [comment]
Thanks @klaraward, I have added invoker.properties, switched to using loadMojo, and update the groupId in the test pom.xml's to be `com.spotify.fmt`

**Review by caesar-ralf — APPROVED** [review]

**Pull request merged**